lib/libzfs_core/libzfs_core.c

```diff
@@ -667,8 +671,13 @@ recv_impl(const char *snapname, nvlist_t *props, const char *origin,
 		(void) strlcpy(zc.zc_name, fsname, sizeof (zc.zc_value));
 		(void) strlcpy(zc.zc_value, snapname, sizeof (zc.zc_value));
 
-		if (props != NULL) {
-			packed = fnvlist_pack(props, &size);
+		allprops = fnvlist_alloc();
+		if (props != NULL)
+			fnvlist_add_nvlist(allprops, "props", props);
```

Comment:
Well, I guess this is going to confuse older kernels anyway...  @behlendorf can weigh in on the right strategy here.